src/format/Kdbx4Reader.cpp

```diff
@@ -224,6 +224,23 @@ bool Kdbx4Reader::readHeaderField(StoreDataStream& device, Database* db)
         variantBuffer.open(QBuffer::ReadOnly);
         QVariantMap data = readVariantMap(&variantBuffer);
         db->setPublicCustomData(data);
+
+        auto it = data.constFind(AUTHENTICATION_FACTORS_HEADER_KEY);
```

Comment:
This is the second change to the existing database reading code: read, validate, and store authentication factor info from one member of the outer header.